set_vcf_sample_id/cloud/run_set_vcf_sample_id.py

```diff
@@ -0,0 +1,237 @@
+      --zones <gce-zones> \
+      --disk-size <size-in-gb> \
+      --original-sample-id <original-id> \
+      --new-sample-id <new-id> \
+      --script-path <gcs-script-path> \
+      --input <gcs-input-path> \
+      --output <gcs-output-path> \
+      --logging <gcs-logging-path> \
+      --poll-interval <interval-in-seconds>
+
+Where the poll-interval is optional (default is no polling).
+
+Users will typically want to restrict the Compute Engine zones to avoid Cloud
+Storage egress charges. This script supports a short-hand pattern-matching
+for specifying zones, such as:
+
+  --zones "*"                # All zones
+  --zones "us-*"             # All US zones
+  --zones "us-central1-*"    # All us-central1 zones
+
+an explicit list may be specified, space-separated:
+  --zones us-central1-a us-central1-b
+
+Passing the --original-sample-id is optional. If set, then the pipeline
+script will verify the value in the input VCF, and if not equal, the
+pipeline will fail.
+
+Note that the pipeline API does not allow for input arguments with no
+value. Thus if the --original-sample-id is not specified (or is empty),
+the ORIGINAL_SAMPLE_ID input parameter is left out of the pipeline definition.
+"""
+
+import argparse
+import pprint
+
+from oauth2client.client import GoogleCredentials
+from apiclient.discovery import build
+
+from pipelines_pylib import defaults
+from pipelines_pylib import poller
+
+# Parse input args
+parser = argparse.ArgumentParser()
+parser.add_argument("--project", required=True,
+                    help="Cloud project id to run the pipeline in")
+parser.add_argument("--disk-size", required=True, type=int,
+                    help="Size (in GB) of disk for both input and output")
+parser.add_argument("--zones", required=True, nargs="+",
+                    help="List of Google Compute Engine zones (supports wildcards)")
+parser.add_argument("--original-sample-id", required=False,
+                    help="The original sample ID to be validated in the input")
+parser.add_argument("--new-sample-id", required=True,
+                    help="The new sample ID")
+parser.add_argument("--script-path", required=True,
+                    help="Cloud Storage path to script file(s)")
+parser.add_argument("--input", required=True, nargs="+",
+                    help="Cloud Storage path to input file(s)")
+parser.add_argument("--output", required=True,
+                    help="Cloud Storage path to output file (with the .gz extension)")
+parser.add_argument("--logging", required=True,
+                    help="Cloud Storage path to send logging output")
+parser.add_argument("--poll-interval", default=0, type=int,
+                    help="Frequency (in seconds) to poll for completion (default: no polling)")
+args = parser.parse_args()
+args.script_path.rstrip('/')
+
+# Create the genomics service
+credentials = GoogleCredentials.get_application_default()
+service = build('genomics', 'v1alpha2', credentials=credentials)
```

Comment:
This is helpful, but maybe I'm misreading something as I was hoping the service account would have pre-configured scopes.  So based on the following, a service account does not seem to have custom scopes attached to it automatically via the Console before creating an instance first - (I'm not referring to the default one):

https://cloud.google.com/compute/docs/authentication#using

https://cloud.google.com/storage/docs/authentication?hl=en#service_accounts

The idea being to be able to have a Builder Pipeline which will create and run a collection of pipelines, such as running dynamically-generated scripts to perform an ensemble analysis for picking the best analysis approach.  Thus the service account or private key that initiated the Builder Pipeline would have `full_control` scope for all APIs (i.e. Pipeline, Storage, Genomics, BigQuery, etc.) before the instances are dynamically created.  This way the authentication process would be transparent, and users just focus on doing the work and the credentials are automatic.  Can these APIs be enabled for domain-wide authority?

~p